sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/rules/RuleExecutor.scala

```diff
@@ -59,10 +59,15 @@ class PlanChangeLogger[TreeType <: TreeNode[_]] extends Logging {
     if (!newPlan.fastEquals(oldPlan)) {
       if (logRules.isEmpty || logRules.get.contains(ruleName)) {
         def message(): MessageWithContext = {
+          // scalastyle:off line.size.limit
           log"""
              |=== Applying Rule ${MDC(RULE_NAME, ruleName)} ===
              |${MDC(QUERY_PLAN, sideBySide(oldPlan.treeString, newPlan.treeString).mkString("\n"))}
+             |
+             |Output Information:
+             |${MDC(QUERY_PLAN, sideBySide(oldPlan.treeString(verbose = false, printOutputColumns = true), newPlan.treeString(verbose = false, printOutputColumns = true)))}
```

Comment:
nit:
```
val oldPlanStringWithOutput = ...
val newPlanStringWithOutput
log"""
   ...
   ... sideBySide(oldPlanStringWithOutput,  newPlanStringWithOutput)
```

Reply:
Can we update the PR description to reflect this change?

Reply:
Good point. Didn't realize the message() method itself is lazily evaluated.

Reply:
Also updated the PR description.